src/process/process_set.cu

```diff
@@ -34,36 +51,29 @@ namespace micm
         yield_offset = 0;
         for (std::size_t i_rxn = 0; i_rxn < n_reactions; ++i_rxn)
         {
-          double rate = rate_constants[i_rxn * n_grids + tid];
-          for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react)
-            rate *= state_variables[reactant_ids_[react_id_offset + i_react] * n_grids + tid];
-          for (std::size_t i_react = 0; i_react < number_of_reactants_[i_rxn]; ++i_react)
+          double rate = device->rate_constants[i_rxn * n_grids + tid];
```

Comment:
Silly question: what is the difference between `->` and `.` operator? Do we have to use `->`?

Reply:
It's a pointer dereference. I was wondering if this part was smart. It may be better to save an alas to each data structure needed a the top of the loop so that we aren't dereferencing a pointer every other instruction, but it's possible that the compiler is smart enough to optimize this out.

Reply:
changes made. I stored the pointers that is being deference multiple times to a pointer variable.